test/Factory/FactoryClassGeneratorTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace ZendTest\Expressive\Tooling\Factory;
+
+use PHPUnit\Framework\TestCase;
+use Zend\Expressive\Tooling\Factory\FactoryClassGenerator;
+use ZendTest\Expressive\Tooling\Factory\TestAsset\ComplexDependencyObject;
+use ZendTest\Expressive\Tooling\Factory\TestAsset\InvokableObject;
+use ZendTest\Expressive\Tooling\Factory\TestAsset\SimpleDependencyObject;
+
+class FactoryClassGeneratorTest extends TestCase
+{
+    /**
+     * @var FactoryClassGenerator
+     */
+    private $generator;
+
+    public function setUp()
+    {
+        $this->generator = new FactoryClassGenerator();
+    }
+
+    public function testCreateFactoryCreatesForInvokable()
+    {
+        $className = InvokableObject::class;
+        $factory = file_get_contents(__DIR__ . '/TestAsset/factories/InvokableObject.php');
+
+        self::assertEquals($factory, $this->generator->createFactory($className));
```

Comment:
Just for consistency, everywhere in this PR you are using `$this->` ... and maybe we can make a final decision what we are gonna do: `self::` or `$this->` (or `static::` 🤣 )

Reply:
I'll fix that; not sure why I used `self::` here; likely copy-pasta from the original in zend-servicemanager.